### DISCUSSION TITLE
Do not purge graphQL cache for specific sections

### DISCUSSION THREAD

**Discussion by edwwaarrdd** [craftcms/cms#9897] [Q&A] [answered]
Now when a section/entry type is created, updated or deleted the whole graphQL cache gets cleared.
Is there a way to skip this behaviour for a specific section/entry type?

**Comment by brandonkelly** [comment, marked as answer]
The more specific your query is, the less likely its cache will get cleared when an entry is saved/deleted.

- If the query includes the `id` argument, it will only get cleared when that specific entry ID is saved/deleted.
- If the query includes the `type`/`typeId` arguments, it will only get cleared when entries of the same type are saved/deleted.
- If the query includes the `section`/`sectionId` arguments, it will only get cleared when entries of the same section are saved/deleted.

It’s not possible to opt a query cache getting cleared in those cases, however.

**Reply by edwwaarrdd** [reply]
Hey @brandonkelly thanks for answering! But the docs says ```Query results are cached to speed up subsequent queries. The GraphQL result cache is very simple, so if the site’s content or structure changes, the entire cache is invalidated.``` (from https://craftcms.com/docs/3.x/graphql.html#caching)
So is it that simple? or is it as you say? :D maybe to avoid confusion I am talking specifically about GraphQL

**Reply by andris-sevcenko** [reply]
@edwwaarrdd the docs need to be updated in this regard. It used to be very simple, but then it was made a bit smarter and now it operates as @brandonkelly says. Essentially, the content part of the cache is managed by the element query cache, while, if anything affecting the schema changes, then the entire cache is purged.

**Reply by edwwaarrdd** [reply]
Okay perfect thanks then its already working as expected 👍 🚀
